test/edu/stanford/dlss/wowza/TestSulWowza.java

```diff
@@ -0,0 +1,836 @@
+        WMSProperties mockProperties = mock(WMSProperties.class);
+        when(mockProperties.getPropertyStr("stacksURL", SulWowza.DEFAULT_STACKS_TOKEN_VERIFICATION_BASEURL)).thenReturn(badUrlStr);
+        IApplicationInstance appInstanceMock = mock(IApplicationInstance.class);
+        when(appInstanceMock.getProperties()).thenReturn(mockProperties);
+        testModule.onAppStart(appInstanceMock);
+        assertEquals(badUrlStr, SulWowza.stacksTokenVerificationBaseUrl);
+
+        try
+        {
+            testModule.onAppStart(appInstanceMock);
+            String logMsg = out.toString();
+            assertThat(logMsg, allOf(containsString("ERROR"),
+                                     containsString(testModule.getClass().getSimpleName()),
+                                     containsString("unable to initialize"),
+                                     containsString("java.net.MalformedURLException"),
+                                     containsString("no protocol")));
+        }
+        finally
+        {
+            logger.removeAppender(appender);
+        }
+    }
+
+    @Test
+    public void onHTTPMPEGDashStreamingSessionCreate_calls_authorizeSession_ifValidConfiguration()
+    {
+        SulWowza spyModule = spy(testModule);
+        HTTPStreamerSessionCupertino sessionMock = mock(HTTPStreamerSessionCupertino.class);
+        spyModule.invalidConfiguration = false;
+        spyModule.onHTTPCupertinoStreamingSessionCreate(sessionMock);
+
+        verify(spyModule).authorizeSession(sessionMock);
+    }
+
+    @Test
+    public void onHTTPMPEGDashStreamingSessionCreate_rejectsSession_ifInvalidConfiguration()
+    {
+        testModule.invalidConfiguration = true;
+        SulWowza spyModule = spy(testModule);
+        HTTPStreamerSessionCupertino sessionMock = mock(HTTPStreamerSessionCupertino.class);
+        testModule.onHTTPCupertinoStreamingSessionCreate(sessionMock);
+
+        verify(sessionMock).rejectSession();
+        verify(spyModule, never()).authorizeSession(sessionMock);
+    }
+
+    @Test
+    public void onHTTPCupertinoStreamingSessionCreate_calls_authorizeSession_ifValidConfiguration()
+    {
+        SulWowza spyModule = spy(testModule);
+        HTTPStreamerSessionMPEGDash sessionMock = mock(HTTPStreamerSessionMPEGDash.class);
+        spyModule.invalidConfiguration = false;
+        spyModule.onHTTPMPEGDashStreamingSessionCreate(sessionMock);
+
+        verify(spyModule).authorizeSession(sessionMock);
+    }
+
+    @Test
+    public void onHTTPCupertinoStreamingSessionCreate_rejectsSession_ifInvalidConfiguration()
+    {
+        testModule.invalidConfiguration = true;
+        SulWowza spyModule = spy(testModule);
+        HTTPStreamerSessionMPEGDash sessionMock = mock(HTTPStreamerSessionMPEGDash.class);
+        spyModule.onHTTPMPEGDashStreamingSessionCreate(sessionMock);
+
+        verify(sessionMock).rejectSession();
+        verify(spyModule, never()).authorizeSession(sessionMock);
+    }
+
```

Comment:
for these four methods, it appears that the names and the test functionality are swapped (the test methods named for `Cupertino` stuff test `MPEGDash` stuff, and vice versa).